docs/_reference/use-cases/ai-app.md

```diff
@@ -0,0 +1,72 @@
+---
+title: Zapier integration structure for an AI app
+order: 4
+layout: post-toc
+redirect_from: 
+
+---
+
+# Zapier integration structure for an AI app
+
+AI app integrations built on Zapier allow users to automate tasks using AI capabilities. Here are some common pain points and recommendations when building AI apps on Zapier.
+
+## Pain Points
+### 1. Long-Running Actions  
+
+**Pain Point:** Actions that take a long time to process can cause issues, especially with Zapier's execution time limits. Test steps in the Zapier editor are limited to 50 seconds, and live Zap executions default to 45 seconds.
+
+**Recommendations:**
+
+* Use `z.generateCallbackUrl()` and `performResume`
+  * `z.generateCallbackUrl()`: This method generates a callback URL that your service can call once the long-running task is complete.  
+  * `performResume`: This function allows the action to pause until the callback URL is called, resuming once the task is complete.  
+  * **Why:** These tools help handle tasks that exceed Zapier's execution time limits by offloading the wait to an external service and resuming only when the task is done [Source].
+* Contact Zapier Partner Support:  
+  * If you need more time for live Zap executions, you can request an extension by contacting Zapier Partner Support through their [contact form](https://zapier.com/partner/contact/). However, we can only do this so much and the most foolproof way would be to implement the previous recommendation.
```

Comment:
discussion in Slack thread if this is offered

Reply:
Removed this based on Slack discussions